### PR TITLE
Define Angular Services for Account Delete

### PR DESCRIPTION
**Pull request by olehyev138**
Delete account pop-up component created.

One action button added to the action buttons display.
The button is disabled unless some keys are selected.

An action button on the table added also.

On Delete component confirmation for the delete is required

### CODE REVIEW

File: src/app/modules/core/mocks/index.ts [hunk 1]
```diff
@@ -117,6 +117,7 @@ export const generateBalancesForEpoch = (url: string) => {
 export const Mocks: IMocks = {
   '/v2/validator/wallet/recover': {},
   '/v2/validator/accounts/exit': {},
+  '/v2/validator/wallet/delete': {},
```
Comment (rauljordan) [inline src/app/modules/core/mocks/index.ts:120]: Let's do /v2/validator/accounts/delete

File: src/app/modules/core/services/wallet.service.ts [hunk 2]
```diff
@@ -101,4 +102,8 @@ export class WalletService {
   exitAccounts(request: AccountVoluntaryExitRequest): Observable<any> {
     return this.http.post(`${this.apiUrl}/accounts/exit`, request);
   }
+
+  deleteAccounts(request: DeleteAccountsRequest): Observable<any> {
+    return this.http.post(`${this.apiUrl}/wallet/delete`, request);
```
Comment (rauljordan) [inline src/app/modules/core/services/wallet.service.ts:107]:
```suggestion
    return this.http.post(`${this.apiUrl}/accounts/delete`, request);
```

File: src/app/modules/wallet/components/account-delete/account-delete.component.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
+
+<div mat-matDialogTitle>
+    <h4>
+        Delete selected account/'s
```
Comment (rauljordan) [inline src/app/modules/wallet/components/account-delete/account-delete.component.html:4]:
```suggestion
        Delete Selected Accounts
```

File: src/app/modules/wallet/components/account-delete/account-delete.component.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            <input
+                matInput
+                formControlName="confirmation"
+                placeholder="Type in agree"
+                name="confirmation"
+                type="text"
+                />
+            <mat-hint>
+                <span>
+                    Type <i>'agree'</i> if you want to exit the keys
```
Comment (rauljordan) [inline src/app/modules/wallet/components/account-delete/account-delete.component.html:37]:
```suggestion
                    Type <i>'agree'</i> if you want to delete the selected keys
```

File: src/app/modules/core/mocks/index.ts (hunk 1 above)
Comment (olehyev138) [reply src/app/modules/core/mocks/index.ts]: fixed

File: src/app/modules/wallet/components/account-delete/account-delete.component.html (hunk 4 above)
Comment (olehyev138) [reply src/app/modules/wallet/components/account-delete/account-delete.component.html]: all fixed

**Comment by rauljordan** [comment]
Hi @olehyev138 

This doesn't look that great. I wonder perhaps we can hide the delete accounts button unless at least one is selected? Also perhaps we can make it fit into one line if we shorten the length of the filter search bar.

Screen Shot 2021-03-12 at 4 03 45 PM

Additionally, the reason we have the "actions" column with a "..." icon is so that actions can be added there. We can add exit account and delete account capabilities in that menu under "view on beaconcha.in"

File: src/app/modules/wallet/components/account-delete/account-delete.component.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                    Type <i>'agree'</i> if you want to delete the selected keys
+                </span>
+            </mat-hint>
+            <mat-error
+                *ngIf="confirmGroup.controls.confirmation.hasError('required')">
+                Confirmation text is required
+            </mat-error>
+            <mat-error
+                *ngIf="confirmGroup.controls.confirmation.hasError('incorectValue')">
+                You must type agree
```
Comment (rkapka) [inline src/app/modules/wallet/components/account-delete/account-delete.component.html:46]:
```suggestion
                You must type 'agree'
```

File: src/app/modules/wallet/components/account-actions/account-actions.component.html [hunk 6]
```diff
@@ -21,4 +21,10 @@
       <mat-icon>exit_to_app</mat-icon>
     </button>
   </a>
+  <a class="ml-3">
+    <button class="large-btn ml-1" *ngIf="selection?.selected?.length >0"
+      color="warn"
+      (click)="openDelete()"
+      mat-raised-button>Delete Accounts</button>
```
Comment (rkapka) [inline src/app/modules/wallet/components/account-actions/account-actions.component.html:28]:
```suggestion
      mat-raised-button>Delete Selected Accounts</button>
```
@rauljordan Do you agree we should add the word `Selected` here? Other buttons have nothing to do with selection, hence the explicitness.

**Review by rkapka — APPROVED** [review]
Looks good to me

**Pull request merged**